Improvements to publishing with automatic PR creation

mosteo opened the pull request:
Add a few steps to the assistant that will end in the PR being opened without having to upload the manifest manually.

From the user PoV, this PR boils down to using `alr publish --submit`.

This is still lacking tests as we may want to tweak the workflow and in any case it will require some more instrumentation to allow dry runs and local impersonation of the community index, and there's already quite some amount of new code. So I would submit the tests in a follow-up.

Having now used the GitHub REST API, it will be easy to improve on this core functionality to allow status tracking and maybe some limited interaction (like requesting a review once checks pass) also from the command line (something like `alr publish --request-review | --status | --cancel`), so for the common case it could all be managed through `alr`. 

This should both simplify submissions and also some people have claimed for automation of the process (people already involved in packaging for distros).

For example, I created https://github.com/alire-project/alire-index/pull/840 with this.

mosteo committed:
Added --submit switch to `alr publish`

mosteo committed:
Forking of community index

mosteo committed:
Changes up to pushing the manifest

mosteo committed:
PR creation

mosteo committed:
Reshaped as steps of the assistant

mosteo committed:
Use release-specific branch

mosteo committed:
Report PR webpage after creation

mosteo committed:
Detect existing conflicting PR

mosteo committed:
Minor tweaks (draft mode, texts, busy spinners)

mosteo committed:
Documentation

mosteo committed:
Allow overriding of community index info

This will allow our testing and help maintainers of other indexes

mosteo committed:
Fix temp dir being not absolute

mosteo committed:
Fixes for self-build with `alr build`

mosteo committed:
Fixes for Windows temp file specifics

Fabien-Chouteau reviewed (approved):
That looks promising :)

Two comments:
- I would love to see a comment explaining the whole process somewhere in the code. It looks fairly complex between the clone, commit and API calls.
- I think this should be a question at the end of `alr publish` rather than a switch.

mosteo committed:
Document all steps more thoroughly

Also use a better name for the new child package.

mosteo committed:
Make submission the default after user confirmation

Document this change in behavior and the new `--skip-submit` switch that
restores the former behavior.

mosteo committed:
Default to submit, and add switch not to submit

mosteo commented on index/pr/prolog/prolog-1.0.0.toml:
```diff
@@ -0,0 +1,20 @@
+website = "https://github.com/jquorning/Prolog"
+tags = ["prolog", "logic"]
+
+project-files = ["prolog_demo.gpr", "prolog_test.gpr"]
```
I see in the repo that the main project file is just `prolog.gpr`, so these shouldn't really be listed here as they're not directly useful or required by clients of the crate. The `prolog.gpr` one can be given explicitly or, if `project-files` is removed, it will be the one used by default.

Ideally, these could be referenced in nested crates, but that's not mandatory for publishing.

jquorning replied:
project-files removed.

mosteo reviewed (changes requested):
This is great, I loved Prolog back in the day.

Just a minor issue to fix with the project files.
